flink-ml-python/pyflink/ml/lib/param.py

```diff
@@ -155,6 +155,48 @@ def input_cols(self) -> Tuple[str, ...]:
         return self.get_input_cols()
 
 
+class HasCategoricalCols(WithParams, ABC):
+    """
+    Base class for the shared input cols param.
+    """
+    CATEGORICAL_COLS: Param[Tuple[str, ...]] = StringArrayParam(
+        "categorical_cols",
+        "Categorical column names.",
+        None,
+        ParamValidators.non_empty_array())
```

Comment:
It might be better to align this parameter with that in Java API. Java `HasCategoricalCols` does not have this validator.